app/controllers/flood_risk_engine/errors_controller.rb

```diff
@@ -0,0 +1,52 @@
+require_dependency "flood_risk_engine/application_controller"
+
+module FloodRiskEngine
+  class ErrorsController < ApplicationController
+
+    def show
+      render(
+        template: file_for(template),
+        locals: { message: exception.try(:message), enrollment: enrollment }
+      )
+    end
+
+    protected
+
+    def error
+      @error ||= params[:id]
+    end
+
+    def template_exists(name)
+      File.exist? template_path(name)
+    end
+
+    def template_path(name)
+      File.expand_path(
+        "app/views/#{file_for(name)}.html.erb",
+        FloodRiskEngine::Engine.root
+      )
+    end
+
+    def template
+      @template ||= template_exists(error) ? error : "generic"
+    end
+
+    def file_for(name)
+      "flood_risk_engine/errors/error_#{name}"
+    end
+
+    def exception
+      env["action_dispatch.exception"]
+    end
+
+    def enrollment
+      return unless journey_token
+      Enrollment.find_by(token: journey_token)
+    end
+
+    def journey_token
+      @journey_token ||= journey_tokens.last
+    end
+
+  end
+end
```

Comment:
A minor thought is that the individual small methods used to determine and format the template path could be in a separate class (pseudo code below):
```
class ErrorsController < ApplicationController

    def show
      template = ExceptionTemplate.new(code: error_code, exception: exception)
      render(
        template: template.path
        locals: { message: exception.try(:message), enrollment: enrollment }
      )
    end

    protected

    class ExceptionTemplate
      def initialize(code:, exception:)
        @code = code
        @exception = exception
      end

      def path
         template_path(@code)
      end

      private

      def template_exists(name)
        File.exist? template_path(name)
      end

      def template_path(name)
        File.expand_path(
          "app/views/#{file_for(name)}.html.erb",
          FloodRiskEngine::Engine.root
        )
      end

      def template
        @template ||= template_exists(error) ? error : "generic"
      end

      def file_for(name)
        "flood_risk_engine/errors/error_#{name}"
      end
    end

```